### ISSUE TITLE
Why boost converter?

### ISSUE THREAD

**Issue by martinjaeger** [philippedc/Arduino-Uno-Wind-Turbine-MPPT-Regulator#2]
Hey, very cool project!

Recently I also thought about a wind turbine MPPT, but I considered a buck converter (battery at low voltage side, wind turbine at higher voltage side) as most suitable DC/DC converter topology, at least for 12V / 24V systems.

Can you please tell a bit more about expected real-life voltage levels reached by the turbine according to your experience? Would we loose much of the energy if we use a buck converter and can't charge the battery at input voltages below battery voltage?

And is stated current in the document meant for the input or output side?

**Comment by philippedc** [comment]
Hi @martinjaeger  I kwew a little about DC-DC converter, but nothing about wind turbine regulators. All literature I read was about PV regulators, they are all buck. But the power answer of a PV is NOT like a wind turbine, even if they are similar. So as I did not know how to do the very first experimentation have been made with a buck-boost inverter converter, with the idea that if the input is too low the boost will make it reach the battery voltage, and if too high the buck....

But what I've seen that this regulator, despite its low efficiency due to the way it works, it was ALWAYS working in boost. May be it could work in buck with a big storm I never get, so, what I can say is that in usual windy environment only boost operates.
So I built a high efficient boost converter. And what I've noticed is that the regulator can operate and gives 10A under 18V. At the beginning I thought my 24V Piggott coils were not well calculated. But not : if the MPPT has found that I can get 180W under 18V that will give 10A, that means I will not get 7.5A under 24V (for the same power). If the wind increases, the voltage will increase according the the batteries charge. And so until the batteries are completely charged at 31V.
So what I can say by my practical experimentation is that the boost converter is not made to increase the output voltage, but instead to decrease the input voltage that increase the current.
VpriMax must be set to the maximum battery voltage, for instance 31V. Then exceed power will go to the Dumpload.

**Comment by martinjaeger** [comment]
I read a bit more about existing wind turbine converters and probably understood why boost makes more sense. Here are my thoughts, correct me if I'm wrong:

The open circuit voltage (voltage without any load) is proportional to the turbine speed. So without an MPPT controller, the wind turbine voltage rises proportional to its rotor speed until it reaches the battery voltage and current starts to flow. The layout of the turbine must be such that the maximum power point is reached around the nominal battery voltage under most typical operation conditions.

In order to optimize the efficiency with an MPPT (i.e. DC/DC converter), there are 3 options:

1. **Boost converter (Vturbine < Vbat)**
This setup allows to generate electricity even at very low speeds where the generator voltage did not yet reach battery voltage. As soon as we approach the battery voltage, the MPPT algorithm doesn't work anymore and the turbine voltage is fixed to the battery voltage. This seems to make sense especially for off-grid systems where it's necessary to get high efficiency when there is not much energy available. If we waste a bit of energy when the turbine generates a high amount of energy anyway, this might not be a big issue.

2. **Buck converter (Vturbine > Vbat)**
In order to make this concept work, it makes sense to use a turbine with higher nominal voltage than the battery system, so e.g. use a 48V turbine with a 24V battery system. This allows to start charging at comparably low speeds already and reach the MPPT in any operation point at higher loads, only limited by the maximum voltage/power of the converter.
However, the voltage under open circuit conditions (e.g. if the battery is full) might get very high, which needs to be considered in the electronics design. A dump load could be used to limit the voltage, but as far as I understood, Hugh Piggott turbines are sometimes operated without a dump load as the turbines are mechanically protected from damage at high wind speeds.
In a grid tied system I guess this approach makes most sense, as you can feed in the energy even if the batteries are already full.

3. **Buck/boost converter**
Combines advantages of both above approaches, but also makes things more complicated + expensive.

**Comment by philippedc** [comment]
Yes, you are right, Except for the advantage of the buck as I do not see the advantage in grid tie configuration. Of course in boost also I feed power when batteries are full!
Buck could make sense in very stormy country and very poor need of power. But I will have to brake the speed in that case.

**Comment by martinjaeger** [comment]
> Of course in boost also I feed power when batteries are full!

Yes, but with the boost converter you can't do MPPT anymore after the the generator voltage reached the battery voltage, which is always the case in very windy conditions. So in this case you don't get the highest efficiency, whereas with the buck layout you could still track the MPP.
What I wanted to say is that in off-grid applications efficiency at low wind is more important and it doesn't matter if you can't do MPPT anymore when you have lots of power/energy available anyway.

**Comment by philippedc** [comment]
Yes you're right. Nevertheless I seceded to reach 700W under 24A with my little 350 W Piggott wind turbine, and I never got those results with the past Chinese regulator.

**Comment by rinntimo** [comment]
What kind of coil you have in your DC/DC converter?
If I undestood correctly the Arduino with Pin 3 offers only 490 Hz PWM output?
Pins 5 and 6 would give 980 Hz but is it enough either?
When I simulated the circuit it seems that relatively high inductance value is needed. I consider 100mH or more for this 490 Hz. 
I have a wind turbine of "2kW" and it gives relatively high voltage when it starts rotating. With a speed of 70 rpm (little over 1 round per second) it gives already 100 Vdc unloaded, rectified. Therefore I have been looking for PPT (Peak Power Tracker) for the wind turbine. I already had MPPT for solar panels, but as you wrote that MPPT works wery different than PPT for wind. It might work somehow without the PPT but I think I will loose quite a lot potential when the high current is breaking down the turbine and it stay on very low power coefficient area. So for my wind turbine I make a Buck (StepDown) converter. The challege is the relatively high voltage on the turbine, as with 260 rpm it gives already 400Vdc. In some studies it has been noted that for 3 blade horizontal axis turbine the highest Cp is found at tip-speed-ratio of 6-7. 
![image](https://user-images.githubusercontent.com/58031832/72546958-6213c700-3894-11ea-9787-ab96d85ca919.png)
I my turbine this situation should achieve already in 5 m/s wind speed -> rotating speed of 170 rpm (which would give unloaded rectified voltage of 260Vdc). 
So I need the Buck anyway. The battery I have for now is 24V so the "full power" current to empty batteries (~20Vdc) is close to 100A. That 260 rpm is so high speed that there is no use to let it rotate faster, so that can be the maximum speed thus the maximum voltage is 400V.
I consider higher frequency for the switching speed.
Any thoughts?

**Comment by philippedc** [comment]
Pwm is 490hz by default with atmega328p, but it is possible to modify registries for a higher frequency. The regulator works with a cutting frequency of 31khz.

**Comment by philippedc** [comment]
In the manual I give the link where I 've bought the coil.

**Comment by rinntimo** [comment]
Thank you for your answers!
☺

Br,
Timo Rinne

**Comment by pulledbeef** [comment]
Hey,
I am quite new to MPPT and find your project quite well documented. Thanks.
I still have a questions concerning the mppt with the boost converter. Lets say I have a 12V load (battery or something else). So the boost converter shall convert whatever voltage beneath 12V comes from the rectifiert to 12V, right. So your T1 Transistor of your DC/DC converter will be triggered that 12V are established, right? Coming from the MPPT perspective the transistor should be triggered that the maximal power is retrieved from turbine. How are both tasks managed with one DC/DC converter? Hope my problem with understanding it right is understandable?

Thanks

**Comment by philippedc** [comment]
@pulledbeef hi, as your working voltage is 12V, the DC-DC converter will work from about 3V to 12V. If your wind turbine gives 14V, the output will be 14V (a little less because of the lost in the rectifier diode). Boost is surprising many people, except if you are considering that voltage and current delivered by the wind turbine are floating. P=V*I, and the only fixed value is Vout = 12V. So the boost effect with a fixed output voltage will adjust the Vin for a best P.
Instead of solar panels a wind turbine is much more a current (in Ampere) generator, that is why a buck converter does not work.

**Comment by pulledbeef** [comment]
Hey, thanks for the quick feedback.
I still dont fully understand you.
What do you mean by voltage and current are floating?
Using a boost converter to deliver a fixed 12V output will result that the pwm of the transistor will be adjusted depending on Vin, right? But how are you controlling the Vin with the boost converter? Sorry for not getting it

**Comment by rinntimo** [comment]
Hi @philippedc and @pulledbeef !
I am not quite sure what you mean the buck converter does not work for wind turbine.
The available power from the wind is relative to cubed wind speed (power of 3).
Therefore the MPPT is used for adapting the input voltage (rotation speed) so that you get the maximum power to use. Here are the possible power curves for different wind seeds and the idea of tracking the maximum power point. (It doesn't matter how you control it; in big wind turbines usually the pitch control is used) 
![kuva](https://user-images.githubusercontent.com/58031832/79535108-26e88600-8085-11ea-9295-df3792c0db32.png)
If you have only "straight connection" from rectifier to battery (as most of the small wind charge controllers basicly do) there are 2 faults:
1) You get no power output until the generator voltage output is above the battery voltage.
2) After you have reached the battery voltage you are limited in power by the electrical characteristic of the generator; internal resistance and maximum current affects, and the aerodynamical charasteristics of the wind turbine it self; it can not give you more power than the increase of torque (= increase of current) output can give; you have fixed the voltage that kind of fixes also the rotation speed. Increasing current will increase the torque and the maximum torque is limited when you run very low tip speed ratio.
Here you see the torque curve shapes of a wind turbine with different wind speeds:
![kuva](https://user-images.githubusercontent.com/58031832/79535764-0faa9800-8087-11ea-9633-e822b6f675cf.png)
If you stay on the maximum torque you loose power as well (as the dotted line is still the same maximum power curve) and if your voltage (~rotation speed) is limited you can not even climb up to the maximum torque point but you are loosing more power.
With the boost converter only you can fix the "fault 1" in the list above and your power curve is like this:
![kuva](https://user-images.githubusercontent.com/58031832/79535040-f0127000-8084-11ea-8789-7c9f6575a4ad.png)
This is some real measurements both in wind tunnel and real world:
![kuva](https://user-images.githubusercontent.com/58031832/79539533-3240af00-808f-11ea-9871-70a6ad385a16.png)

Of cource you can double your maximum power output if you have a "12V generator" rated 350W and you double the battery voltage you can get 700W out of it (because of the maximum current it can deliver is what it is).
If you have most of the winds so small that the rotation speed gets very seldom above the "battery voltage speed" then the boost converter only is fine. If you have also bigger wind speeds you should use a buck-boost converter to get the most out of it. But that is much more difficult to control...

As I understand the permanent magnet generator still behaves like "voltage source": if you increase speed you get more voltage. I understand the difference to solar panels behavior, where the maximum voltage can be much higher than the "maximum power output voltage" - it has relatively high internal impedance. Then the increase of current with the same solar input gives you rapidly less power because the output voltage drops easily when you increase load. Thats what the solar MPPT does; adapts the weak input impedance to the rigid impedance of a battery.
My wind turbine is quite big as a small wind turbine; blade length is about 1,5 m. And the generator voltage rises quite rapidly. I don't need the boost option as it reaces the voltage of a 24V battery right when it starts to rotate, but without a buck converter I will loose almost 90% of the potential power output.
Of course at "maximum voltage" (or unreasonable speed) you have to brake it down first by increasing the load current and at last shorting the windings.
I will try to convert this Arduino project to control a bigger Buck converter for my wind turbine, as the voltage can get more than 200V with that ludicrous rotation speed and the rated power of the generator is something like 2kW with a huge ability of delivering current output.

As an answer to @pulledbeef question: how to control the "input voltage" with the DC/DC converter;
If you increase the current it increases torque and depending the turbine and the wind speed that affects the rotation speed more or less, and the "input voltage" (= generator output voltage) is affected because of that. Hope the curves (I found from the Internet by Google picture search) clarifies this a bit more....

**Comment by pulledbeef** [comment]
Wow,
thanks for the details.
Looking at the boost scenario I still cant fully understand it:
Let's say we have a 12V Battery. The turbine is producing 10V freerunning. Using a Boost converter we can boost the input voltage of the turbine to the battery voltage of 12V. As soon as we would start using the boost converter the turbine Voltage would decrease as we are putting some load on it. As the boost converter will be triggered in order to reach the fixed 12V of the battery, how can it also be used to put the wind turbine in optimal power point? The pwm for triggering the boost converter will be set so that 12V or whatever loading voltage for the battery is set. For example we have a wind of 3m/s and the optimum power of the windturbine would be at 8V, how can we put the windturbine into that working spot with the same boost converter? 

In my understanding you would need two dc/dc converters, one for putting the turbine in the right power point, resulting in an unfixed output voltage. And a second converter using the output voltage of the first to convert to the desired battery loading voltage

**Comment by philippedc** [comment]
@rinntimo many thanks for your explanations, you do it better I could.

Note that my first prototype was a buck boost inverter DC DC converter. Dispite the low efficiency of this kind of converter, I noticed it never worked in buck mode. But always in boost, or with a ratio of 0.
Even if those tests have not been done during a storm, this can suppose that the turbine voltage optimal operation is  below Vmax, eg 31V for my 24V batteries set.

My only experience are with Piggott wind turbines, 3 different sizes. So... But, if you can reach "easily" 200V, may be your turbine is more appropriate for a 48V or 96V battery set, than a 24V.

**Comment by rinntimo** [comment]
@philippedc You are welcome!
Yes my wind turbine/generator might be good straight to 96V battery system, but for convenience I prefer 24V batteries for many device is easier to find using straight 24V or 12V after my 24V-to-12V voltage adapter / 12V battery charger... Any way then I would need the Boost converter for small winds. And it is easier to find 2 similar 12V batteries to put in series than 4 or 8... when the battery pack needs to be enhanced/replaced. :)

@pulledbeef Lets put it this way: The Boost converter is not _setting_ the output voltage to 12V because it already is that, if you are connected to a 12V battery. It is a matter of "pumping" charge through current flow from the generator (lower voltage level) to the battery (fixed but higher voltage level), like an in-house air-heat-pump is doing in air conditioning device.
Also this animation (first 5 minutes) might get you some idea: [https://www.youtube.com/watch?v=vwJYIorz_Aw](https://www.youtube.com/watch?v=vwJYIorz_Aw)
Here they have a battery with "rigid" voltage at the input and a variable voltage to the output load, but in wind generator-to-battery -case it is just the opposite. Anyway the MPPT Boost converter is draining as much current as possible from the generator to the battery, so that the generators power output 
P = U x I 
is maximum possible. As the generators voltage drops when you drain more current (~torque) than the power from the wind at that moment is allowing you to drain. So it is kind of adjusting the rotating speed as well.

**Comment by pulledbeef** [comment]
@philippedc 
thanks dude. Yeah the point I was missing is indeed that the battery is setting the voltage. Thanks

**Comment by MafuLynch** [comment]
> @rinntimo many thanks for your explanations, you do it better I could.
> 
> Note that my first prototype was a buck boost inverter DC DC converter. Dispite the low efficiency of this kind of converter, I noticed it never worked in buck mode. But always in boost, or with a ratio of 0.
> Even if those tests have not been done during a storm, this can suppose that the turbine voltage optimal operation is below Vmax, eg 31V for my 24V batteries set.
> 
> My only experience are with Piggott wind turbines, 3 different sizes. So... But, if you can reach "easily" 200V, may be your turbine is more appropriate for a 48V or 96V battery set, than a 24V.

Hi, I have been looking into making a MPPT controller myself too. My understanding of how it works and why it only owrked in boost for you is because of the turbine desing. When you read piggott books he calculates the ammount of turns in the coil and rotor diameter to get the rated voltage at your desired speed. For it to work as buck you should need it to be much higher. I am looking to make a 48v wind turbine, so following piggott recepie I should do 4 coils of 90 turns, but I am looking to make 5 coils at 150 turns each, this way the rated voltage will increase to 100V aprox when loaded. This means in low wind I would get 48V with low rotating speed and as wind increases voltage will increase too. 
I am still analyzing the advantage of doing this mode vs a boost converter. I have read several IEEE papers about turbine MPPT controller and they all talk about using buck converters only. I believe the reason is you can get more power out of it, since the current will opose the wind torque on the turbine. So at same current, same oposing torque but with higher voltage you get higher electrical power. 

So what I  am trying to say is there is no right way going for boost or buck converter, but it depends on the design of the turbine since with one that was designed with to operate at 48V would need too high wind to maintain a proper TSR and produce high voltage. 
Will keep posted on any progress I manage to do.

Your proyect has been very helpfull for me to orientate my ideas

**Comment by philippedc** [comment]
hi @MafuLynch, it would be great to get your source about IEEE papers about buck wind turbine. 

Since now I am experimenting the regulator for a pair of years. The code has been improved (v3, not published yet  because a luck of time to update the manual) with the help of several contributors, thank to them.

They also shared other models of wind turbine regulators, for instance this one : https://mousa-simple-projects.blogspot.com/2020/07/mppt-wind-charge-controller-reverse.html
It is also a boost controller.

**Comment by MafuLynch** [comment]
@philippedc I will attach some of the papers I have read. I have several more pending to read still. 

This is not an IEEE paper, but it has some interesting info in how they increased the efficiency of the buck converter
[751830.pdf](https://github.com/philippedc/Arduino-Uno-Wind-Turbine-MPPT-Regulator/files/6572325/751830.pdf)

[soetedjo2011.pdf](https://github.com/philippedc/Arduino-Uno-Wind-Turbine-MPPT-Regulator/files/6572343/soetedjo2011.pdf)

[matecconf_icudr2019_03004.pdf](https://github.com/philippedc/Arduino-Uno-Wind-Turbine-MPPT-Regulator/files/6572348/matecconf_icudr2019_03004.pdf) this one also wasn't IEEE but also interesting. 

I will go through all the colection of papers I have and sort them out and share them. Because lots of them had some nice information but not quite applicable to these type of turbines and I can't identify now the ones I liked.

But it is my understanding that both methods are possible. I also have a paper somewhere where thay are using a boost too.

**Comment by MafuLynch** [comment]
Let me share with you also this comment on Hugh Piggott book.

"So far the assumption has been that the voltage is clamped by a battery. You can also consider changing the operation voltage in response to changes in speed of a given alternator. This is sometimes called 'maximum power point tracking' (MPPT). It's quite an exciting avenue of wind turbine development. 
In low winds, the alternator can only generate relatively low voltage, whereas in high winds it has the ability to produce much higher voltage. Connecting it directly to the battery handicaps its efficiency by tying it to the same voltage under all coonditions. Using a converter to change the voltage allowing it to rise in higher winds, has several benefits.

- The alternator losses depend on the square of the current, so an increase in voltage and corresponding reduction in current (for a given power output) cuts the losses in the coils, saving energy and avoiding overheating.
- The blades will work best at a constant tip speed ratio which implies that the rotational speed should change in direct proportion to windspeed. Working a constant voltage implies working at a more like a constant speed, whereas using a converter allows you to vary the alternator speed more widely.

...

The basic direct battery-charging design without a converter has poor efficiency in stronger winds because of the high resistance in the stator(caused by the need to cutt in at low rpm) and also the tendecy to stall the blades (for the same reason)..."

He just compares using DC-DC converters vs direct  connecting to the battery, but he only talks about going up in voltage. 

I am by no means a wind energy expert, but I understand out of this that if the turbine is designed to generate 48V at low rpm (which would depend on the ammount of turns per coil, number of coils and strength of magnet) will allow to go higher and have a better efficiency. I am still to find a paper that compares the performance of using low voltage and rising to high voltage vs producing high voltage and lowering it. I would find this very interesting and would like to do the research for it at some point.

But I really admire your work. My comment was going to that I think that the difference in using a boost or buck also depends on the turbine design since you might need too high wind to reach high voltages if it was not designed to generate those high voltages. 

In his website Hugh also talks about some people in france using 5 coils per phase with 260 turns pero coil, allowing it to go much higher in voltage.
